src/module.c

```diff
@@ -4990,7 +5127,7 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         }
         acl_retval = ACLCheckAllUserCommandPerm(ctx->client->user,c->cmd,c->argv,c->argc,&acl_errpos);
         if (acl_retval != ACL_OK) {
-            sds object = (acl_retval == ACL_DENIED_CMD) ? c->cmd->name : c->argv[acl_errpos]->ptr;
+            sds object = (acl_retval == ACL_DENIED_CMD) ? (sds)c->cmd->name : c->argv[acl_errpos]->ptr;
```

Comment:
did we change the type? how did this work till now?

Reply:
it used to be `char *` and now it's `const char *`

Reply:
But is it an sds? if not then this casting is invalid. 
If yes, then maybe it should be typed correctly.

Reply:
it's sds only for module commands, const char * fo native commands

Reply:
ok, a little bit awkward. i hope it doesn't bite us some day.

Reply:
addACLLogEntry calls sdsnew not matter what, i think we should change it to take `const char*`

WDYT?

Reply:
or, considering the function takes an sds, it should have used `sdsdup` (faster)

Reply:
yes, but that would cause a crash now

either we make sure it gets a real a sds or we decide it gets a const char*

your call

Reply:
if it takes an sds, and someone passes a non-sds, should crash, and we need to either change the arg type or fix the caller.
but why would it crash?
the code we're looking at already has an sds, is the same object (command) used elsewhere as a char*?
anyway, please try to fix it, if you can make it a real sds and use sdsdup with minimal changes, let's do it.
but if in many cases we have to create an sds, just in order to dup it, let's use char*
so please try to sort it up and decide which one is better.